lib/src/main/kotlin/bstrees/RBTree.kt

```diff
@@ -7,8 +7,17 @@ class RBTree<T : Comparable<T>> : SelfBalancingBST<T, RBNode<T>>() {
     override fun createNewNode(data: T) = RBNode(data)
     override val balancer = RBBalancer<T>()
 
+    /**
+     * Insert data in tree.
+     *
+     * @param data the type of data (should be comparable)
+     * @return Nothing
+     */
```

Comment:
I don't see any point in using `@param`, `@return` statements here.
I suggest you rewrite the documentation comment as following
```suggestion
    /** Inserts new node with [data] as its value in the tree */
```